.github/workflows/ci.yml

```diff
@@ -151,12 +156,16 @@ jobs:
       - name: Checkout code
         uses: actions/checkout@v4
 
+      - name: Remove Infection requirement if not needed
+        if: matrix.run-mutation-tests != 'yes'
+        run: composer remove --dev infection/infection --no-update
+
       - name: Setup PHP with Composer deps
         uses: ./.github/composite-actions/setup-php-with-composer-deps
         with:
           os: ${{ runner.os }}
           php: ${{ matrix.php-version }}
-          php-coverage: ${{ matrix.collect-code-coverage }}
+          php-coverage: ${{ matrix.collect-code-coverage || matrix.run-mutation-tests }}
```

Comment:
questio: do i understand right that we need code-coverage for running mutation tests?

Reply:
Yes. It can be generated by Infection (current usage) or passed as argument if pre-generated

Reply:
got you, thanks ;)